src/LinuxTracingIntegrationTests/LinuxTracingIntegrationTest.cpp

```diff
@@ -1049,5 +1068,68 @@ TEST(LinuxTracingIntegrationTest, ModuleUpdateOnDlopen) {
   EXPECT_TRUE(module_update_found);
 }
 
+TEST(LinuxTracingIntegrationTest, GpuJobs) {
+  if (!CheckIsStadiaInstance()) {
+    GTEST_SKIP();
+  }
+  if (!CheckIsPerfEventParanoidAtMost(-1)) {
+    GTEST_SKIP();
+  }
+  LinuxTracingIntegrationTestFixture fixture;
+
+  std::vector<orbit_grpc_protos::ProducerCaptureEvent> events =
+      TraceAndGetEvents(&fixture, PuppetConstants::kVulkanTutorialCommand);
+
+  VerifyOrderOfAllEvents(events);
+
+  bool another_process_used_gpu = false;
+  for (const auto& event : events) {
+    if (event.event_case() != orbit_grpc_protos::ProducerCaptureEvent::kFullGpuJob) {
+      continue;
+    }
+
+    const orbit_grpc_protos::FullGpuJob& gpu_job = event.full_gpu_job();
+    if (gpu_job.pid() != fixture.GetPuppetPid()) {
+      another_process_used_gpu = true;
+      break;
+    }
+  }
+  LOG("another_process_used_gpu=%d", another_process_used_gpu);
+
+  uint64_t gpu_job_count = 0;
+  for (const auto& event : events) {
+    if (event.event_case() != orbit_grpc_protos::ProducerCaptureEvent::kFullGpuJob) {
+      continue;
+    }
+
+    const orbit_grpc_protos::FullGpuJob& gpu_job = event.full_gpu_job();
+    if (gpu_job.pid() != fixture.GetPuppetPid()) {
+      continue;
+    }
+
+    // The puppet is single-threaded.
+    EXPECT_EQ(gpu_job.tid(), fixture.GetPuppetPid());
+
+    EXPECT_LT(gpu_job.amdgpu_cs_ioctl_time_ns(), gpu_job.amdgpu_sched_run_job_time_ns());
+    // If no other job is running on the GPU (which is the case if the puppet is the only process
+    // using the GPU), then we assume (it's the best we can do) that the job starts running on the
+    // hardware at the same time as it is scheduled by the driver, hence the EXPECT_EQ.
+    // Otherwise, use EXPECT_LE.
```

Comment:
Are we sure about this?

Reply:
Are you referring to a specific part?
The idea here is that the tutorial is always only running one job at a time (offscreen rendering might be playing a role in this, I'm not sure how the full tutorial would behave). 
When no other job is running on the gpu, we assume it runs immediately after being scheduled and the middle time interval has 0 duration.
We check if we can make this assumption for every single job by seeing if there were no jobs from other processes while we were tracing.
Do you see some part that could be problematic in this?

Reply:
>we assume it runs immediately after being scheduled and the middle time interval has 0 duration.

I am not sure. I was basically wondering about the confidence that this is true. Couldn't there be a super tiny amount of time until the second trace point gets hidden, even if there is nothing else on the gpu?

Reply:
No no, this is our own assumption, that we make in `GpuTracepointVisitor.cpp`. The problem here is that we have three tracepoints for four time points, so this is the best we can do.

From the code:
```
  // We assume that GPU jobs (command buffer submissions) immediately
  // start running on the hardware when they are scheduled by the
  // driver (this is the best we can do), *unless* there is already a
  // job running. We keep track of when jobs finish in
  // timeline_to_latest_dma_signal_. If a previous job is still running
  // at the timestamp of scheduling the current job, we push the start
  // time for starting on the hardware back.
```
and
```
  // We do not have an explicit event for the following timestamp. We
  // assume that, when the GPU queue corresponding to timeline is
  // not executing a job, that this job starts exactly when it is
  // scheduled by the driver. Otherwise, we assume it starts exactly
  // when the previous job has signaled that it is done. Since we do
  // not have an explicit signal here, this is the best we can do.
```

Reply:
Ahhh. Thanks for the clarification.

Reply:
In other words, the first, second, and fourth time point come from a tracepoint, while the third we have to assume based on what else is happening on the gpu. When nothing else is running on the gpu, we assume that the second and third time points coincide.